### DISCUSSION TITLE
Component Style and __css prop bug?

### DISCUSSION THREAD

**Discussion by keithburgie** [chakra-ui/chakra-ui#7309] [Q&A] [answered]
In the [Theming/Component Style docs](https://chakra-ui.com/docs/styled-system/component-style#return-values), it shows the following example:

<img width="695" alt="Screenshot 2023-02-01 at 12 50 32 PM" src="https://user-images.githubusercontent.com/96078669/216122894-04eaa692-a0c5-48e3-b6e2-b78d6460a687.png">

I'm having trouble where it shows `<Flex __css={styles.menu} {...rest}>` When I use the `__css` prop, it unsets all styles. If I use `Flex` or `Stack` or anything else, it renders as a `Box` with `display: block`.

I've copied directly from the example into [a Code Sandbox here](https://codesandbox.io/s/friendly-booth-ondxs0?file=/src/index.tsx) and see the same issue. The `Flex` component loses `display: flex`:

<img width="602" alt="Screenshot 2023-02-01 at 1 06 42 PM" src="https://user-images.githubusercontent.com/96078669/216126412-2272ddf0-60e6-4536-8f6e-0a5a0255eda3.png">

Am I missing something? Can I extend any components besides `Box`? If not, it's a real bummer to have to redefine everything.

**Comment by keithburgie** [comment, marked as answer]
I've answered my own question. It works correctly if you cast the `Box` as another component.

```
// 🟢 Works: Uses `display: flex`
<Box as={Flex} __css={styles.menu} {...rest}>
  <StylesProvider value={styles}>{children}</StylesProvider>
</Box>

// 🟢 Works: Uses `display: flex` and applies margin (via spacing prop) to children
<Box as={Stack} __css={styles.menu} {...rest}>
  <StylesProvider value={styles}>{children}</StylesProvider>
</Box>


// 🔴 Doesn't work: Returns a div with `display: block`
<Flex __css={styles.menu} {...rest}>
  <StylesProvider value={styles}>{children}</StylesProvider>
</Box>
```

**Reply by ankursehdev** [reply]
I was seeing the same issue, thanks for the solution!
